src/Microsoft.HttpRepl/Commands/TreeNode.cs

```diff
@@ -15,7 +15,9 @@ public class TreeNode
         private readonly List<TreeNode> _children = new List<TreeNode>();
 
         public TreeNode(Formatter formatter, string prefix, string entry)
-            : this(formatter, prefix, entry, 0)
+            : this(formatter ?? throw new ArgumentNullException(nameof(formatter)),
```

Comment:
Seems like the null check should be a concern for the other constructor.  Even though it's private, if an e.g. factory method were added, it ought to have the same checks applied.

Reply:
I thought I had tried that and it was still throwing the Analyzer error on the public constructor. But I just tried it again and it seems to be ok with it. Maybe I hadn't fully refreshed the analyzer results (sometimes VS seems to not clear the list).